### PR TITLE
RAV-9189:IBC_Add Support for Qualify in big query

### PR DESCRIPTION
**Pull request by Shubham-Isalkar**
IBC : Add Support for Qualify in big query

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -2043,6 +2045,52 @@ List<SqlNode> getListOfCaseOperands(SqlCase sqlCase) {
...
+          } else {
+            return getRexInputRefFromCondition(operand);
+          }
+        }
+      }
+      return condition;
+    }
+
+    /** Returns whether an Analytical Function is present in filter condition. */
+    private boolean hasAnalyticalFunctionPresentInFilter(RelNode rel) {
```
Comment (sudatdm) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:2064]: Can we rename it to `hasAnalyticalFunctionInFilter`?

Comment (sudatdm) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:2064]: Param type should be `Filter` Rel if its for filter only.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 2]
```diff
@@ -2043,6 +2045,52 @@ List<SqlNode> getListOfCaseOperands(SqlCase sqlCase) {
...
+              }
+            }
+          }
+        }
+      }
+      return false;
+    }
+
+    private boolean hasAnalyticalFunctionPresentInProjection(Project projectRel) {
+      return projectRel.toString().contains("ROW_NUMBER()");
```
[the remark this answers is not in the document]
Comment (Shubham-Isalkar) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: Previously, it was discussed that this defect needs to be handled for ROW_NUMBER() function only.

Now, I have implemented the generic approach to handle all analytical functions.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 3]
```diff
@@ -2043,6 +2045,52 @@ List<SqlNode> getListOfCaseOperands(SqlCase sqlCase) {
...
+                return true;
+              }
+            }
+          }
+        }
+      }
+      return false;
+    }
+
+    private boolean hasAnalyticalFunctionPresentInProjection(Project projectRel) {
```
[the remark this answers is not in the document]
Comment (Shubham-Isalkar) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: This hasAnalyticalFunction is already used. Have used the 2nd one.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java (hunk 1 above)
Comment (Shubham-Isalkar) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: Done, renamed.

Comment (Shubham-Isalkar) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: Done

**Review by deepakkadam-dm — APPROVED** [review]
Looks Ok

**Pull request merged**